### DISCUSSION TITLE
[Suggestion] SnapStop (AutoStop)

### DISCUSSION THREAD

**Discussion by herovaxy** [cafali/SnapKey#62] [New Features & Ideas]
I'm not sure it's bannable or not, correct me if I'm wrong. But essentially, AutoStop allows the players to press and release the key in order to stop the movement. The best example is the game Valorant where press and release is common practice.

Recently, I've been seeing a video made by Tripp on YT where the AutoStop config allows players to perform this in-game. Perhaps you can add the feature SnapStop?

Instead of having another key to perform the counter strafe, the moment player release the key... Then, it'll auto-strafe for you.

Here's the config on CS2:
`alias "+autostop_forward" "+forward; rightleft 0 1 0; !forwardback 0 1 0"
alias "-autostop_forward" "-forward; !forwardback 0.00000000000001 0 0"
alias "+autostop_back" "+back; rightleft 0 1 0; forwardback 0 1 0"
alias "-autostop_back" "-back; forwardback 0.00000000000001 0 0"
alias "+autostop_left" "+left; forwardback 0 1 0; rightleft 0 1 0"
alias "-autostop_left" "-left; rightleft 0.00000000000001 0 0"
alias "+autostop_right" "+right; forwardback 0 1 0; !rightleft 0 1 0"
alias "-autostop_right" "-right; !rightleft 0.00000000000001 0 0"

bind "w" "+autostop_forward"
bind "s" "+autostop_back"
bind "a" "+autostop_left"
bind "d" "+autostop_right"`

YouTube: https://www.youtube.com/watch?v=PKAU70vWce0

**Comment by minteeaa** [comment]
so, essentially a bind that taps the other directional key on release of the first, without user input of the second?

i believe it's implementable but could be really stretching the scope of what "snap tap" should be;

at that point, you are actively having a program input the opposite keypress to stop a strafe without user input of the opposite key - this itself i feel is a lot more along the lines of "cheating" than anything else

that being said - if this gets implemented into Razer or Wooting's firmware/software as a feature, SnapKey (might) aim to maintain parity with those - assuming it's within scope

**Comment by sdj88** [comment]
Hi chipping in to let OP know that this is most definitely more on the cheating side because it's pre-inputting a movement you haven't done yet and in the early days of 1.6 and CSS people used methods such as this to create bunny hop scripts and defensive scripts like you fire an Awp shot and it automatically flicks to knife and executes a bunnyhop backwards to avoid being shot. Some of them even went as far as implementing mouse commands so that if you held m1 with a rifle it would adjust recoil after 4/5 shots to x and y for compensation. 

I bring that to your attention as trying to make that happen in that valve are very aware of movement scripts and they only allowed the jump throw bind in faceit/mm because it allows more casual players to throw a lineup... they arent allowed in tournaments. 

Lastly a feature like what you want might work in some situations ingame like entry fragging on a wide swing on T's however those movement type scripts when used back in the day people didnt really like them for the fact that 1v1's you need to be able to jiggle A+D for micro adjustments say like your heads exposed behind the back box of D2 on B site... and having a feature that pre-emptively uses commands will actually give you less control in your duels.


EDIT***  

I just watched the video and the top comment says what I said in the above.

Pinned by Tripp

guys dont actually use this if you wanna play competitvely, its not viable. you cannot go up ladders, you cannot airstrafe. its cooked. its a proof of concept nothing more, if youre really bad at counter strafing you can put this on a toggle with a hotkey in the console.

**Comment by stvnkng** [comment]
i wish there was snap stop
